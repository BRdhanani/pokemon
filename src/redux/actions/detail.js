import ApiRequestService from '../../services/api.service';
import DetailActionTypes from '../types/detail';
import {loaderEnd, loaderStart} from './loader';

export const getPokemonSpecies = name => async dispatch => {
  try {
    await dispatch(loaderStart());
    const res = await ApiRequestService.getApi(`/pokemon-species/${name}`);
    let str = '';
    res.data.flavor_text_entries.map(flavor => {
      str = str + flavor.flavor_text;
    });
    dispatch({
      type: DetailActionTypes.GET_POKEMON_SPECIES,
      payload: str,
    });
    await dispatch(loaderEnd());
  } catch (error) {
    dispatch(loaderEnd());
  }
};

export const getPokemonEgggroup = name => async dispatch => {
  try {
    await dispatch(loaderStart());
    const res = await ApiRequestService.getApi(`/egg-group/${name}`);
    dispatch({
      type: DetailActionTypes.GET_POKEMON_EGGGROUP,
      payload: res.data.name,
    });
    await dispatch(loaderEnd());
  } catch (error) {
    dispatch(loaderEnd());
  }
};
