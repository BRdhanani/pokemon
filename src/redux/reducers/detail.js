import DetailActionTypes from '../types/detail';

const initialState = {
  pokemonDetail: '',
  egg: '',
};

export default (state = initialState, action) => {
  switch (action.type) {
    case DetailActionTypes.GET_POKEMON_SPECIES:
      return {
        ...state,
        pokemonDetail: action.payload,
      };
    case DetailActionTypes.GET_POKEMON_EGGGROUP:
      return {
        ...state,
        egg: action.payload,
      };
    default:
      return state;
  }
};
