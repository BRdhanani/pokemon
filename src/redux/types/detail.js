const DetailActionTypes = {
  GET_POKEMON_SPECIES: 'GET_POKEMON_SPECIES',
  GET_POKEMON_GENDER: 'GET_POKEMON_GENDER',
  GET_POKEMON_EGGGROUP: 'GET_POKEMON_EGGGROUP',
};

export default DetailActionTypes;
