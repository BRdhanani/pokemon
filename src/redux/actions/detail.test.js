import ApiRequestService from '../../services/api.service';
import {getPokemonEgggroup, getPokemonSpecies} from './detail';

jest.mock('../../services/api.service');

describe('detail actions', () => {
  beforeEach(() => jest.resetAllMocks());

  it('should get PokemonSpecies', () => {
    const dispatcher = getPokemonSpecies('bulbasaur');

    dispatcher(jest.fn());

    expect(ApiRequestService).toMatchSnapshot();
  });

  it('should get PokemonEgggroup', () => {
    const dispatcher = getPokemonEgggroup('bulbasaur');

    dispatcher(jest.fn());

    expect(ApiRequestService).toMatchSnapshot();
  });
});
