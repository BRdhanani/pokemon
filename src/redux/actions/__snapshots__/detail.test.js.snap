// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detail actions should get PokemonEgggroup 1`] = `
ApiRequestService {
  "axiosService": [MockFunction],
  "baseURL": "https://pokeapi.co/api/v2",
  "constructor": [MockFunction],
  "getApi": [MockFunction],
}
`;

exports[`detail actions should get PokemonSpecies 1`] = `
ApiRequestService {
  "axiosService": [MockFunction],
  "baseURL": "https://pokeapi.co/api/v2",
  "constructor": [MockFunction],
  "getApi": [MockFunction],
}
`;

exports[`detail actions should get bookmarks 1`] = `
ApiRequestService {
  "axiosService": [MockFunction],
  "baseURL": "https://pokeapi.co/api/v2",
  "constructor": [MockFunction],
  "getApi": [MockFunction],
}
`;
